src/displayapp/screens/Tile.cpp

```diff
@@ -123,9 +133,25 @@ void Tile::UpdateScreen() {
   lv_label_set_text(batteryIcon, BatteryIcon::GetBatteryIcon(batteryController.PercentRemaining()));
 }
 
-void Tile::OnValueChangedEvent(lv_obj_t* obj, uint32_t buttonId) {
+void Tile::OnValueChangedEvent(lv_obj_t* obj, uint32_t buttonId, lv_event_t event) {
   if(obj != btnm1) return;
+  switch(event){
+    case LV_EVENT_VALUE_CHANGED:
+        app->StartApp(apps[buttonId], DisplayApp::FullRefreshDirections::Up);
+        running = false;    
+      break;
+    case LV_EVENT_LONG_PRESSED:
+      currentFavoriteApp = settingsController.GetFavoriteApp();
+      if(buttonId){
```

Comment:
buttonId is the value from 0 to 5 of the current button on screen so this won't allow selection of the first button

Reply:
I realize that, however, all that is supposed to be done is to save the selected application as favorite - basically the same parameters are used to actually launch the application on a single tap. I don't understand how saving the application would be any different, as the types are ( or should be ) the same.